chromium_src/chrome/browser/ui/webui/top_chrome/webui_contents_wrapper.cc

```diff
@@ -8,44 +8,42 @@
 #include <utility>
 
 #include "components/sessions/content/session_tab_helper.h"
-#include "components/site_engagement/content/site_engagement_helper.h"
-#include "content/public/browser/web_contents.h"
-#include "content/public/browser/web_contents_delegate.h"
-#include "content/public/browser/web_contents_observer.h"
 
-// The buble delegate doesnt allow to open popups and we use
+#define BRAVE_WEBUI_CONTENTS_WRAPPER_ADD_NEW_CONTENTS                          \
+  if (!browser_delegate_) {                                                    \
+    return host_                                                               \
+               ? host_->AddNewContents(                                        \
+                     source, std::move(new_contents), target_url, disposition, \
+                     window_features, user_gesture, was_blocked)               \
+               : nullptr;                                                      \
+  }                                                                            \
+  auto* raw_popup_contents = new_contents.get();                               \
+  auto* contents = browser_delegate_->AddNewContents(                          \
+      source, std::move(new_contents), target_url,                             \
+      WindowOpenDisposition::NEW_POPUP, window_features, user_gesture,         \
+      was_blocked);                                                            \
+  if (!contents) {                                                             \
+    return nullptr;                                                            \
+  }                                                                            \
+  auto tab_id = sessions::SessionTabHelper::IdForTab(raw_popup_contents).id(); \
+  popup_ids_.push_back(tab_id);                                                \
+  return contents;
+
+#include "src/chrome/browser/ui/webui/top_chrome/webui_contents_wrapper.cc"
+
+#undef BRAVE_WEBUI_CONTENTS_WRAPPER_ADD_NEW_CONTENTS
+
+// The bubble delegate doesn't allow to open popups and we use
```

Comment:
don't these comments belong to the code above?

Reply:
Good point, moved via https://github.com/brave/brave-core/pull/26068/commits/779e7196e0be31d8a8e49a4bb5ee970352e798de

Reply:
all of this code is really concerning in general, I'm going to open a follow-up ticket to look at it...